test/t8_schemes/t8_gtest_find_parent.cxx

```diff
@@ -58,60 +76,32 @@ t8_recursive_child_find_parent (t8_element_t *element, t8_element_t *child,
   }
 }
 
-static void
-t8_compute_child_find_parent (const int maxlvl)
-{
-  t8_element_t       *element, *child, *test_parent;
-  t8_scheme_cxx      *scheme;
-  t8_eclass_scheme_c *ts;
-  int                 eclassi;
-  t8_eclass_t         eclass;
-  scheme = t8_scheme_new_default_cxx ();
-  for (eclassi = T8_ECLASS_ZERO; eclassi < T8_ECLASS_COUNT; eclassi++) {
-    t8_productionf ("Testing child find parent for eclass %s\n",
-                    t8_eclass_to_string[eclassi]);
-    eclass = (t8_eclass_t) eclassi;
-    /* Get scheme for eclass */
-    ts = scheme->eclass_schemes[eclass];
-    /* Get element and initialize it */
-    ts->t8_element_new (1, &element);
-    ts->t8_element_new (1, &child);
-    ts->t8_element_new (1, &test_parent);
-
-    ts->t8_element_set_linear_id (element, 0, 0);
-    /* Check for correct parent-child relation */
-    t8_recursive_child_find_parent (element, child, test_parent, ts, 0,
-                                    maxlvl);
-    /* Destroy element */
-    ts->t8_element_destroy (1, &element);
-    ts->t8_element_destroy (1, &child);
-    ts->t8_element_destroy (1, &test_parent);
-
-  }
-  /* Destroy scheme */
-  t8_scheme_cxx_unref (&scheme);
-}
-
-int
-main (int argc, char **argv)
+TEST_P (class_find_parent, t8_compute_child_find_parent)
 {
-  int                 mpiret;
 #ifdef T8_ENABLE_LESS_TESTS
   const int           maxlvl = 4;
 #else
   const int           maxlvl = 6;
 #endif
 
-  mpiret = sc_MPI_Init (&argc, &argv);
-  SC_CHECK_MPI (mpiret);
-  sc_init (sc_MPI_COMM_WORLD, 1, 1, NULL, SC_LP_ESSENTIAL);
-  t8_init (SC_LP_DEFAULT);
+  t8_element_t       *element;
```

Comment:
These elements should be members of the class. Please move the construction, initialization and the deconstruction of the elements into the class.